File: mpp/msqueue-maxim092001/src/test/java/msqueue/LinearizabilityTest.java
package msqueue;

import org.jetbrains.kotlinx.lincheck.LinChecker;
import org.jetbrains.kotlinx.lincheck.annotations.Operation;
import org.jetbrains.kotlinx.lincheck.strategy.stress.StressCTest;
import org.junit.Test;

@StressCTest(sequentialSpecification = SequentialQueue.class)
public class LinearizabilityTest {
    private final Queue queue = new MSQueue();

    @Operation
    public void enqueue(int x) {
        queue.enqueue(x);
    }

    @Operation
    public int peek() {
        return queue.peek();
    }

    @Operation
    public int dequeue() {
        return queue.dequeue();
    }

    @Test
    public void test() {
        LinChecker.check(LinearizabilityTest.class);
    }
}

File: mpp/universal-construction-maxim092001/test/LinearizabilityTest.kt
import org.jetbrains.kotlinx.lincheck.LinChecker
import org.jetbrains.kotlinx.lincheck.annotations.Operation
import org.jetbrains.kotlinx.lincheck.strategy.stress.StressCTest
import org.jetbrains.kotlinx.lincheck.verifier.VerifierState
import kotlin.test.Test


@StressCTest
class LinearizabilityTest : VerifierState() {
    private val c = Solution()

    @Operation
    fun add(x: Int) = c.getAndAdd(x)

    @Test
    fun testLinearizability() {
        LinChecker.check(LinearizabilityTest::class.java)
    }

    override fun extractState(): Any = c.getAndAdd(0)
}